### PR TITLE
feat(tests): enhance testing framework and guidelines

### PR DESCRIPTION
**Pull request by misama-ct**
## Summary

This PR enhances Nimbus's testing infrastructure by introducing a dual testing strategy that clearly separates unit tests for pure functions from Storybook interaction tests for UI components.

### Two Distinct Testing Approaches

1. **Unit Tests (Vitest)** - Fast, isolated tests for pure functions
2. **Storybook Tests** - Browser-based tests for UI components and interactions

### When to Use Each

**Unit Tests (`*.test.ts`) are for:**
- Pure functions with deterministic outputs
- Data transformations (parsers, formatters, converters)
- Validation logic (type guards, emptiness checks)
- Calculations (financial math, sorting algorithms)
- Security utilities (sanitization, XSS prevention)
- String manipulation (locale sorting, text processing)

**Storybook Tests (`*.stories.tsx` with play functions) are for:**
- UI interactions (clicks, typing, keyboard navigation)
- Component behavior (state changes, rendering)
- Visual regressions (screenshots, layout)
- Accessibility (ARIA attributes, screen reader support)
- Integration (component composition, context providers)

### New Configuration Files

- **`vitest.unit.config.ts`** - Dedicated config for fast unit tests
  - No browser environment overhead
  - Excludes `*.stories.tsx` files
  - Optimized for pure function testing
  - Coverage reporting for utilities

- **`vitest.storybook.config.ts`** - Renamed from `vitest.config.ts`
  - Browser-based testing with Playwright
  - Tests component stories with play functions
  - Chromium headless mode for CI

### Updated Commands

```bash
# Run fast unit tests (new)
pnpm test:unit

# Run unit tests in watch mode (new, for TDD)
pnpm test:unit:watch

# Generate coverage report (new)
pnpm test:coverage

# Run Storybook tests (existing)
pnpm test:storybook

# Run all tests sequentially (new)
pnpm test:all

# Run all tests (parallel, existing)
pnpm test
```

### New Guidelines Document

Added comprehensive **Unit Testing Guidelines** (`docs/file-type-guidelines/unit-tests.md`) covering:

- Clear decision matrix for when to use unit tests vs Storybook tests
- Test structure and organization patterns
- AAA pattern (Arrange, Act, Assert)
- Edge case and error condition testing
- Test data management strategies
- Vitest-specific features and matchers
- Performance testing guidelines
- Security testing patterns
- Best practices and common pitfalls

### Updated Documentation

- **`CLAUDE.md`** - Added testing commands and strategy overview
- **`docs/component-guidelines.md`** - Linked to unit testing guidelines

## 🧪 Example Implementation

Added `noop.test.ts` as a reference implementation demonstrating:
- Basic test structure
- Edge case testing
- JSDoc documentation for test utilities
- Co-location pattern (test next to utility)

## 🎨 Benefits

1. **Faster Feedback Loop** - Unit tests run in milliseconds vs seconds
2. **Clear Separation of Concerns** - Different tools for different jobs
3. **Better Developer Experience** - Watch mode for TDD workflows
4. **Comprehensive Coverage** - Track utility function coverage
5. **CI Optimization** - Fast unit tests can run on every commit
6. **Scalability** - Unit test suite scales better than browser tests

## 🔄 Migration Path

For existing code:
1. Identify pure utility functions in components
2. Extract to dedicated utility files if needed
3. Write unit tests using the new guidelines
4. Keep UI interaction tests in Storybook stories

## ✅ Testing This PR

```bash
# Run the new unit tests
pnpm test:unit

# Run in watch mode
pnpm test:unit:watch

# Generate coverage report
pnpm test:coverage

# Verify Storybook tests still work
pnpm test:storybook
```

## 📖 Related Documentation

- [Unit Testing Guidelines](docs/file-type-guidelines/unit-tests.md)
- [Stories Guidelines](docs/file-type-guidelines/stories.md)
- [Component Guidelines](docs/component-guidelines.md)

---

**Note**: This is a foundational change that establishes patterns for future development. The testing strategy will evolve as we add more utility functions and components to the design system.

### CODE REVIEW

**Review by tylermorrisford — APPROVED** [review]
that's some beautiful documentation

**Review by ByronDWall — APPROVED** [review]
Overall this looks really good, thanks!

Only thing I noticed is the `*.test` vs `*.spec` thing, and my care amount is somewhat low, but I do think we should settle on one file type for the unit tests, and to me it makes sense for it to be the same as used in the rest of ct's frontend repos

**Comment by jaikamat** [comment]
@misama-ct I piggy-backed on this by migrating some of the static method tests from `MoneyInput`'s Storybook test to the unit test framework in 07e39697939a93fa8ed29ecb04b3f3bb00f027a9, worked well. I hope you don't mind me including it

File: pnpm-workspace.yaml [hunk 1]
```diff
@@ -16,57 +15,60 @@ catalogs:
...
     next-themes: ^0.4.6
     react-aria: 3.44.0
     react-aria-components: 1.13.0
     react-stately: 3.42.0
-    '@react-aria/interactions': 3.25.6
-    '@react-aria/optimize-locales-plugin': 1.1.5
-    '@internationalized/date': ^3.10.0
+    "@react-aria/interactions": 3.25.6
+    "@react-aria/optimize-locales-plugin": 1.1.5
+    "@internationalized/date": ^3.10.0
```
Comment (valoriecarli) [inline pnpm-workspace.yaml]: 🤣

Comment (misama-ct) [reply pnpm-workspace.yaml]: This week is double-quote week.
